skbeam/core/accumulators/bin_image.py

```diff
@@ -0,0 +1,219 @@
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import numpy as np
+
+
+class BinImage(object):
+    """
+    Class to bin an ndarray onto an axis using a
+    user-specified bin for each element of the array.
+    """
+
+    def __init__(self, bin_values, norm, nbins, min=None, max=None,
+                 weights=None):
+        """
+        Parameters
+        ----------
+        bin_values : np.ndarray
+            For each pixel, this is the value that determines which bin
+            a pixel's intensity will contribute to.  It can be an arbitrary
+            array of floats/ints which will be discretized into nbins equally
```

Comment:
That might not necessarily be fair. Since the bin_values describes where a pixel is on your histogram while the binning itself describes which pixels go together and/or the resolution of your pixels. For instance the Q resolution of the pixels actually changes as a function of Q.
[res.pdf](https://github.com/scikit-beam/scikit-beam/files/266022/res.pdf) This is an example with a Perkin Elmer detector.